### PR TITLE
[GitHub App] Connections can clone all installation repositories

### PR DESCRIPTION
**Pull request by pjlast**
Adds a config option so that GitHub App installation connections can clone all repositories the App was installed on, and sets that as the default configuration.

GitHub's `/users/:username/repos` api cannot return private repositories, so for user installations of Apps, there's no clean way to clone all the repos.

With this PR, App Installation connections will, with the default setup, clone all repositories the App was installed on, which I also think is intuitive and expected.

## Test plan

Verified manually. Added VCR test.

### CODE REVIEW

File: internal/repos/github.go [hunk 1]
```diff
@@ -590,6 +590,32 @@ func (s *GitHubSource) listUser(ctx context.Context, user string, results chan *
 	return
 }
 
+// listUser returns all the repositories belonging to the authenticated GitHub App installation
```
Comment (sashaostrikov) [inline internal/repos/github.go:593]:
```suggestion
// listAppInstallation returns all the repositories belonging to the authenticated GitHub App installation
```

**Review by sashaostrikov — APPROVED** [review]
LGTM!
Worth a line in our changelog IMO 🤔

**Comment by pjlast** [comment]
@sashaostrikov I'm not sure if it matters for the Changelog, since this is part of an unreleased feature 🤔

**Comment by sashaostrikov** [comment]
> @sashaostrikov I'm not sure if it matters for the Changelog, since this is part of an unreleased feature 🤔 

Ahh, then it makes sense not to add it, right!

**Review by varsanojidan — APPROVED** [review]

File: internal/repos/github.go [hunk 2]
```diff
@@ -1148,6 +1174,10 @@ func (s *GitHubSource) listAllRepositories(ctx context.Context, results chan *gi
 	for i := len(s.config.Orgs) - 1; i >= 0; i-- {
 		s.listOrg(ctx, s.config.Orgs[i], results)
 	}
+
+	if s.config.GitHubAppDetails != nil && s.config.GitHubAppDetails.CloneAllRepositories {
```
[the remark this answers is not in the document]
Comment (pjlast) [reply internal/repos/github.go:1178]: I'll try 😬 will have to see how it plays out with the funky GitHub App auth procedure

**Review by kopancek — APPROVED** [review]

**Pull request merged**